TP-Sampler/src/codap-com.js

```diff
@@ -471,9 +471,84 @@ CodapCom.prototype = {
       selectSelf(this.myCODAPId);
     }
   },
+
   register: function (action, resource, operation, callback) {
     codapInterface.on(action, resource, operation, callback);
-  }
+  },
+
+  sendFormulaToTable: async function (measureName, measureType, selections) {
+    var samplesColl = this.getCollectionNames().samples;
+    function getFormula () {
+      if (measureType === "count") {
+        return "count()";
+      } else if (measureType === "sum") {
+        return `sum(${selections.output})`
+      } else if (measureType === "conditional_count") {
+        console.log("${selections.operator}", selections.operator);
+        console.log(`count(${selections.output}${selections.operator}${selections.value})`);
+        return `count(${selections.output} ${selections.operator} '${selections.value}')`
+      } else if (measureType === "conditional_percentage") {
+        return `100 * count(${selections.output} ${selections.operator}${selections.value}) / count()`
+      } else if (measureType === "mean") {
+        return `mean(${selections.output})`
+      } else if (measureType === "median") {
+        return `median(${selections.output})`
+      } else if (measureType === "conditional_sum") {
+        return `sum(${selections.output}${selections.operator}${selections.value}, ${selections.output2})`
+      } else if (measureType === "conditional_mean") {
+        return `mean(${selections.output}, ${selections.output2}${selections.operator}${selections.value})`
+      } else if (measureType === "conditional_median") {
+        return `median(${selections.output}, ${selections.output2}${selections.operator}${selections.value})`
+      } else if (measureType === "difference_of_means") {
+        return `min(mean(${selections.outputPt1}, ${selections.outputPt12}${selections.operatorPt1}${selections.valuePt1}), mean(${selections.outputPt2}, ${selections.outputPt22}${selections.operatorPt2}${selections.valuePt2}))`
+      } else if (measureType === "difference_of_medians") {
+        return `min(median(${selections.outputPt1}, ${selections.outputPt12}${selections.operatorPt1}${selections.valuePt1}), mean(${selections.outputPt2}, ${selections.outputPt22}${selections.operatorPt2}${selections.valuePt2}))`
+      }
+    }
+
+    codapInterface.sendRequest({
+      action: "get",
+      resource: `${getTargetDataSetPhrase()}.collection[${samplesColl}].attributeList`
+    }).then((res) => {
+      const attrs = res.values;
+      let newAttributeName;
+      // check if attr name is already used. user could add "conditional count" twice, for example,
+      // but have difference formulas (output = a, output = b)
+      const attrNameAlreadyUsed = attrs.find((attr) => attr.name === measureName);
```

Comment:
Is this to append an index number to the attribute name if attribute name is already in use? I'm surprise CODAP doesn't automatically change it since there is existing code in CODAP that detects for this.